compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -794,6 +795,13 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             }
         }
 
+        let def_id = self.def_id.to_def_id();
+        let optimization_disabled = if self.tcx.def_kind(def_id).has_codegen_attrs() {
+            self.tcx.codegen_fn_attrs(def_id).optimize.as_ref().is_some_and(|o| o.is_none())
```

Comment:
Yeah, adding on to the review comment above, this use will look confusing to a reader.